packages/webpack-plugin/src/index.js

```diff
@@ -101,6 +101,19 @@ class LoadablePlugin {
   apply(compiler) {
     this.compiler = compiler
 
+    const { webpack } = compiler
+
+    const defs = {
+      'process.isBrowser':
+        compiler.options.target === 'web' ||
+        compiler.options.target === undefined,
+    }
```

Comment:
To go with my suggestion of adding a `serverSideModuleFederation` argument to the webpack plugin, if we do then this logic could throw an error for misconfiguration:

```suggestion
   if (this.opts.serverSideModuleFederation === true &&
     compiler.options.target === 'web' ||
     compiler.options.target === undefined
   ) {
     throw new Error(`Enabling server-side module federation support for a client-side Webpack target (${compiler.options.target}) is unnecessary and will effectively disable all code-splitting.`
   }
```